src/RuleSet/RuleSets.php

```diff
@@ -67,4 +68,45 @@ public static function getSetDefinition(string $name): RuleSetDescriptionInterfa
 
         return $definitions[$name];
     }
+
+    /**
+     * @param class-string<RuleSetDescriptionInterface> $class
+     */
+    public static function registerRuleSet(string $name, string $class): void
+    {
+        $preg = new Preg();
+
+        if (1 !== $preg->match('/^@[a-zA-Z][a-zA-Z0-9_\/\.-]*$/', $name)) {
+            throw new \InvalidArgumentException('RuleSet name must begin with "@" and a letter (a-z, A-Z), and can contain only letters (a-z, A-Z), numbers, underscores, slashes, dots and hyphens.');
```

Comment:
# alignment needed:

I do believe that we must have a strong indicator that RuleSet is not a built-in ruleset but custom one. We do it for Rules by `name` (built-in) or `vendor/name` (for custom rules)

for me it's must-have to have sth similar for Rulesets too. I would blindly pick `vendor/@ruleset`, but I'm happy to discuss different proposals.

I also agree that `PHPCSFixer` and it's variants should be reserved for us.


btw, for custom rules, we have `final class FixerNameValidator`, why using different approach for RuleSets?

Reply:
I prefer `@vendor/ruleset` because it's easier to spot the `@`.

Reply:
I agree with @Wirone about the naming.

If we want to be super clear about ruleset being custom, we could force it to be called `@Custom/Vendor/Ruleset` or just `@Custom/Ruleset,` but I'm open to your requirements. Maybe this is a bit much.

I think the only reason to not have equivalent of `FixerNameValidator` is that I wasn't aware of it. I'm happy to implement it once we know what we want to do.

Reply:
> I think the only reason to not have equivalent of FixerNameValidator is that I wasn't aware of it.

Exactly like me, but I worked with it during #7064 recently and that's why I discovered issue with the regex here 😉. `RuleSetNameValidator` can be introduced anyway, regardless of final decision, we can just move current regex there. But you can wait until final decision too, it's up to you.

Reply:
ad1 - please be aware that single vendor could provide multiple different sets, you should not assume that there will be only one - it could be the case, but it's not limited to it.

for `vendor/@ruleset` vs `@vendor/ruleset` - let's use one of those formats.

Reply:
I'm voting for `@vendor/ruleset`. If no opposition, I will implement it so that vendor name `@PhpCsFixer*` is reserved, anything else is free game.